sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/WithKeys.java

```diff
@@ -129,6 +132,20 @@ public KV<K, V> apply(V element) {
       // TODO: Remove when we can set the coder inference context.
       result.setCoder(KvCoder.of(keyCoder, in.getCoder()));
     } catch (CannotProvideCoderException exc) {
+      if (keyType != null) {
+        try {
+          SchemaRegistry schemaRegistry = SchemaRegistry.createDefault();
+          SchemaCoder<K> schemaCoder =
+              SchemaCoder.of(
+                  schemaRegistry.getSchema(keyType),
+                  keyType,
+                  schemaRegistry.getToRowFunction(keyType),
+                  schemaRegistry.getFromRowFunction(keyType));
+          result.setCoder(KvCoder.of(schemaCoder, in.getCoder()));
+        } catch (NoSuchSchemaException exception) {
+          // No Schema.
```

Comment:
Shouldn't we fail in this case?

Reply:
I think no, because the coder still could be set manually. There is a comment below: "let lazy coder inference have a try"